core/app/models/spree/order_cancellations.rb

```diff
@@ -42,6 +46,42 @@ def short_ship(inventory_units, whodunnit:nil)
     unit_cancels
   end
 
+  # Marks inventory unit canceled. Allows specifying the reason why.
+  #
+  # @return [UnitCancel] the unit that has been canceled
+  def cancel_unit(inventory_unit, reason: Spree::UnitCancel::DEFAULT_REASON, whodunnit:nil)
+    unit_cancel = nil
+
+    Spree::OrderMutex.with_lock!(@order) do
+      unit_cancel = Spree::UnitCancel.create!(
+        inventory_unit: inventory_unit,
+        reason: reason,
+        created_by: whodunnit,
+      )
+
+      inventory_unit.cancel!
+    end
+
+    unit_cancel
+  end
+
+  # Reimburses inventory units due to cancellation.
+  #
+  # @return [Reimbursement] the reimbursement for items being canceled
+  def reimburse_units(inventory_units)
+    reimbursement = nil
+
+    Spree::OrderMutex.with_lock!(@order) do
+      return_items = inventory_units.map(&:current_or_new_return_item)
+      reimbursement = Spree::Reimbursement.new(order: @order, return_items: return_items)
+      reimbursement.return_items.map(&:accept!)
```

Comment:
This is a bit gross. I would like to be able to call `reimbursement.return_all` which does all we need it to do.